Source/WTF/wtf/JSONValues.h

```diff
@@ -186,10 +199,10 @@ class ObjectBase : public Value {
 protected:
     ~ObjectBase();
 
+    template<typename T>
+    void setPrimitive(const String& name, T&& value) requires std::is_arithmetic_v<std::remove_cvref_t<T>>;
+
     // FIXME: use templates to reduce the amount of duplicated set*() methods.
-    void setBoolean(const String& name, bool);
```

Comment:
at the very least i feel like not having this could make it difficult for callers to use (i.e. implicitly convert) a non-`bool` value

for example, something like `setBoolean("hasFoo", foo())` which returns a pointer